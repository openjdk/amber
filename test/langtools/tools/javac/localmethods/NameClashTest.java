/*
 * @test /nodynamiccopyright/
 * @bug 8233038
 * @summary Tests need to be written for local methods feature
 * @compile/fail/ref=NameClashTest.out -XDrawDiagnostics NameClashTest.java
 */

public class NameClashTest {

    class X{}

    int foo() {
        return 10;
    }
    public static void main(String args []) {
        <T> int foo(T t) {
            return 10;
        }
        int foo(Object o) { // Error after erasure
            return 10;
        }
        if (args == null) {
            int foo(Object o, String s) { return 10; } // OK.
            int foo(Object o) { // Error after erasure
                return 10;
            }
        }

        int foo(String ... s) {
            return 10;
        }

        int foo(String [] s) {  // error varags signature same.
            return 10;
        }

        int foo() { return 10; }  // OK.

        void foo(X x) {}
        void foo(X x1, X x2) {} // OK.
        void foo(X x) {} // clash
    }

    void goo() {
        if (10 == 10) {
            void goo(int x) {} // OK
            void goo() { // OK.
                void goo() {} // Error.
                class Y {
                    void goo() {} // OK.
                }
            }
        } else {
            void goo(int x) {} // OK
            void goo() {} // OK.
        }
        void goo(int x) {} // OK
        void goo() {} // OK.
    }
}
