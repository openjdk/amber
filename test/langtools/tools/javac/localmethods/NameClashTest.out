NameClashTest.java:19:13: compiler.err.name.clash.same.erasure: foo(java.lang.Object), <T>foo(T)
NameClashTest.java:24:17: compiler.err.name.clash.same.erasure: foo(java.lang.Object), <T>foo(T)
NameClashTest.java:33:13: compiler.err.array.and.varargs: foo(java.lang.String[]), foo(java.lang.String...), main(java.lang.String[])
NameClashTest.java:41:14: compiler.err.already.defined: kindname.method, foo(NameClashTest.X), kindname.method, main(java.lang.String[])
NameClashTest.java:48:22: compiler.err.already.defined: kindname.method, goo(), kindname.method, goo()
5 errors
